File: 575.DistributeCandies/Candies.java
class Solution {
    public int distributeCandies(int[] candies) {
        Set<Integer> candySet = new HashSet<>();
        for (int c : candies) {
            candySet.add(c);
        }
        return Math.min(candies.length / 2, candySet.size());
    }
}

File: 575.DistributeCandies/candies.cc
class Solution {
public:
    int distributeCandies(vector<int>& candies) {
        set<int> candySet;
        for (int c : candies) {
            candySet.insert(c);
        }

        return min(candySet.size(), candies.size() / 2);
    }
};
